crate2nix/src/resolve.rs

```diff
@@ -276,10 +276,11 @@ pub fn configured_source_is_used_instead_of_local_directory() {
     let root_package = &indexed.root_package().expect("root package");
 
     let mut crate2nix_json = crate::config::Config::default();
-    let source = crate::config::Source::CratesIo {
+    let source = crate::config::Source::Registry {
         name: "some_crate".to_string(),
         version: semver::Version::from_str("1.2.3").unwrap(),
         sha256: "123".to_string(),
+        index: url::Url::parse("registry+https://github.com/rust-lang/crates.io-index").unwrap(),
```

Comment:
Shoudn't "registry" be a different field (enum maybe) rather than abuse a string?

Reply:
Registry index can be any custom URL, not sure how Enum comes into the picture

Reply:
There's two types of sources, right? crates.io and registries (which themselves have URLs). So instead of having all of that together inside a String (and then see if the string format is like the crates.io format or starts with registry+) that seems like an Enum with two elements would be a lot neater.

Reply:
crates.io is itself a registry

Reply:
fair enough. Well, crates.io with a specific URL and other registries :-) My point is that there's two distinct situations there.